src/engine/enginebufferscale.h

```diff
@@ -65,17 +48,27 @@ class EngineBufferScale : public QObject {
     }
 
     // Set the desired output sample rate.
-    virtual void setSampleRate(int iSampleRate) {
-        m_iSampleRate = iSampleRate;
+    virtual void setSampleRate(SINT iSampleRate);
+
+    const Mixxx::AudioSignal& getAudioSignal() const {
+        return m_audioSignal;
     }
 
     // Called from EngineBuffer when seeking, to ensure the buffers are flushed */
     virtual void clear() = 0;
     // Scale buffer
-    virtual double getScaled(CSAMPLE* pOutput, const int iBufferSize) = 0;
+    // Returns the virtual number of sample frames that have been read.
+    // The actual number of frames copied to the output buffer is always
+    // an integer value, while the virtual number of read frames might be
+    // partial number!
+    virtual double getScaledSampleFrames(
```

Comment:
sample frames is a new name. 
Since the parameter is a pointer to samples and the iBufferSize is a sample count,
getScaledSamples() is correct if we look to the output parameter. 
If we look on the return value it returns the consumed frames from unscaled input buffer. 
getReadFrames(); 
On the other hand this is no classic getter ... mmm

How about: scaleFrames()